### ISSUE TITLE
ENH:dataframe columns dropping with column names case-insensitive

### ISSUE THREAD

**Issue by rmsmani** [pandas-dev/pandas#35796]
#### Is your feature request related to a problem?
NO

#### Describe the solution you'd like
Currently the data frame columns are case-sensitive, when we try to drop the column with different case, the error is thrown.
My suggestion is to have one more option like ignorecase as boolean, if the flag is true, then drop the column if the name satisifies

#### API breaking implications

[this should provide a description of how this feature will affect the API]

#### Describe alternatives you've considered

[this should provide a description of any alternative solutions or features you've considered]
Currently, changing the both the dataframe column and the drop column to lower case and delete it.

#### Additional context
Its good to have feature..

```python
# Your code here, if applicable

import io
import pandas as pd

source = io.StringIO("""col1, col2,col3
1,2,3
""")

df = pd.read_csv(source)

print (df)
df = df.drop("COL3", axis=1)
print(df)

   col1   col2  col3
0     1      2     3

Error
  File "test/test.py", line 16, in <module>
    df = df.drop("COL3", axis=1)

KeyError: "['COL3'] not found in axis"

```

**Comment by alimcmaster1** [comment]
Might be easier to just convert all your column names to upper or lower case first?

**Comment by rmsmani** [comment]
That is how am doing it now.  
It's good to hv this option in dataframe drop function

**Comment by jreback** [comment]
this option would have to be added in quite a few places 

-1 on this expansion as there are plenty of ways to do this now

**Comment by mroeschke** [comment]
I'd be -1 as well as `drop` can take more than just strings so a new argument would be niche

**Comment by jorisvandenbossche** [comment]
I agree that this is probably not something we want to get into (there are many places in pandas where case (in)sensitivity could play a role)

**Comment by jorisvandenbossche** [comment]
(although we have some prior art here, several str methods have a `case` keyword, like [`str.contains()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.contains.html) or [`str.replace()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.replace.html))
